### PR TITLE
deployments(ticdc): move all docker related files into deployments

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve? 
part of https://github.com/pingcap/ticdc/issues/2199

### What is changed and how it works?
1. Move dockerfiles into /deployments/ticdc/docker
2. Move docker-compose files into  /deployments/ticdc/docker-compose
3. rename Dockerfile.development to dev.Dockerfile（So that IDE can recognize it）

### Check List 

Tests 

 - Integration test
 - No code

Code changes

None

Side effects
None

Related changes

None

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
/cc @amyangfei @overvenus 

/hold

When the code review is over, I will revert the trigger mechanism for the corn job.

**Comment by hi-rustin** [comment]
Regarding the integration test code , I will take the time to address it in the next PR, we have a lot of duplicate constants that need to be extracted.

**Comment by hi-rustin** [comment]
The logs can be uploaded!
image

File: .github/workflows/ticdc_integration_cron.yaml [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - 'dm/**'
+  pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - 'dm/**'
```
Comment (amyangfei) [inline .github/workflows/ticdc_integration_cron.yaml:14]: Is this change used for test here?

Comment (amyangfei) [reply .github/workflows/ticdc_integration_cron.yaml:14]: > When the code review is over, I will revert the trigger mechanism for the corn job.

I find the comment.

**Review by amyangfei — COMMENTED** [review]
Is there any CI/CD job using the `Dockerfile`?

File: .github/workflows/ticdc_integration_cron.yaml (hunk 1 above)
Comment (hi-rustin) [reply .github/workflows/ticdc_integration_cron.yaml:14]: Yes!

**Comment by hi-rustin** [comment]
> Is there any CI/CD job using the `Dockerfile`?

I don't think so, the tests all passed properly.

File: .github/workflows/ticdc_integration_cron.yaml (hunk 1 above)
Comment (hi-rustin) [reply .github/workflows/ticdc_integration_cron.yaml:14]: see: https://github.com/pingcap/ticdc/pull/3204#issuecomment-955715717

**Comment by amyangfei** [comment]
> I don't think so, the tests all passed properly.

I mean some release procedure, such as releasing TiCDC docker image

**Comment by overvenus** [comment]
The TiCDC image in dockerhub does not use the docker file in repo.

See https://hub.docker.com/layers/pingcap/ticdc/v5.1.2/images/sha256-df4a7f94c4e982230e3130c76109d7a144c80d069709e612723faf5cb709685d?context=explore

**Review by overvenus — APPROVED** [review]
LGTM

**Review by amyangfei — APPROVED** [review]

**Comment by hi-rustin** [comment]
/assign @amyangfei 

Could you please help me retrigger the merge process? Thanks!

**Pull request merged**